src/lib.rs

```diff
@@ -46,6 +67,200 @@ impl UniffiCustomTypeConverter for Uuid {
+        })
+}
+
+fn convert_to_mrz_entry(s: &[u8]) -> &[u8; 30] {
+    s.try_into().expect("slice with incorrect length")
+}
+
+#[derive(thiserror::Error, uniffi::Error, Debug)]
+pub enum VCBVerificationError {
+    #[error("{value}")]
+    Generic { value: String },
+}
+
+#[uniffi::export]
+pub async fn verify_vcb_qrcode_against_mrz(
+    mrz_payload: String,
+    qr_payload: String,
+) -> Result<(), VCBVerificationError> {
+    let mrz: MRZ = mrz_payload
+        .lines()
+        .map(|x| *convert_to_mrz_entry(x.as_bytes()))
+        .collect::<Vec<[u8; 30]>>()
+        .try_into()
+        .map_err(|_| VCBVerificationError::Generic {
+            value: "Invalid MRZ string".to_string(),
+        })?;
+
+    // First we decode the QR-code payload to get the VCB in CBOR-LD form.
+    let input = MachineReadableZone::decode_qr_code_payload(qr_payload.as_str()).map_err(|e| {
+        VCBVerificationError::Generic {
+            value: e.to_string(),
+        }
+    })?;
+
+    // Then we decompress the CBOR-LD VCB to get a regular JSON-LD VCB.
+    let vc = decode_from_bytes::<MachineReadableZone>(&input)
+        .await
+        .map_err(|e| VCBVerificationError::Generic {
+            value: e.to_string(),
+        })?;
+
+    // Finally we verify the VCB against the MRZ data.
+    let params = VerificationParameters::new(AnyDidMethod::default().into_vm_resolver());
+    verify(&vc, &mrz, params)
+        .await
+        .map_err(|e| VCBVerificationError::Generic {
+            value: e.to_string(),
+        })?
+        .map_err(|e| VCBVerificationError::Generic {
+            value: e.to_string(),
+        })
+}
+
+#[derive(thiserror::Error, uniffi::Error, Debug)]
+pub enum VCVerificationError {
+    #[error("{value}")]
+    Generic { value: String },
+}
+
+#[uniffi::export]
+pub async fn verify_json_vc_string(json: String) -> Result<(), VCVerificationError> {
+    use ssi::prelude::VerificationParameters;
+
+    let vc = any_credential_from_json_str(&json).map_err(|e| VCVerificationError::Generic {
+        value: e.to_string(),
+    })?;
+
+    let vm_resolver = AnyDidMethod::default().into_vm_resolver();
+    let params = VerificationParameters::from_resolver(vm_resolver);
+
+    vc.verify(&params)
+        .await
+        .map_err(|e| VCVerificationError::Generic {
+            value: e.to_string(),
+        })?
+        .map_err(|e| VCVerificationError::Generic {
+            value: e.to_string(),
+        })
+}
+
+#[derive(thiserror::Error, uniffi::Error, Debug)]
+pub enum VPError {
+    #[error("{value}")]
+    Verification { value: String },
+    #[error("{value}")]
+    Parsing { value: String },
+}
+
+#[uniffi::export]
+pub async fn vc_to_signed_vp(vc: String, key_str: String) -> Result<String, VPError> {
+    use ssi::prelude::*;
+
+    let vp = ssi::claims::vc::v1::JsonPresentation::new(None, None, vec![vc]);
+
+    let mut key: ssi::jwk::JWK = serde_json::from_str(&key_str).map_err(|e| VPError::Parsing {
```

Comment:
If you want, you can define a new type like with do for UUID with `uniffi::custom_type!`, but it's not necessary for now

Reply:
I prefer to keep it this way because we are just using it internally. If we export/return this at some point we can create a custom type